### PR TITLE
ENH: support none in header list #37625

### LINKED ISSUE #37625: ENH: support none in header list
I have a file with two sheets, one has a header, and another hasn't. 
The following code will raise `TypeError: list indices must be integers or slices, not NoneType.`

```python
# Your code here, if applicable
df = pd.read_excel(file_path, sheet_name=None, header=[0, None])
```

### PR DESCRIPTION
**Pull request by Alvinwuzw**
closes #37625
tests passed
read_excel funtion can now accept header={'Sheet1': 0, 'Sheet2': None} as dictionary with header for each sheet.

### CODE REVIEW

File: pandas/io/w1902.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import sys
```
Comment (arw2019) [inline pandas/io/w1902.py:1]: What do these two files do?

Comment (Alvinwuzw) [reply pandas/io/w1902.py:1]: Sorry, It was added by mistake! I could remove it right away.

Comment (Alvinwuzw) [reply pandas/io/w1902.py:1]: Should I make another pull request after resolving this?

**Review by arw2019 — CHANGES_REQUESTED** [review]
Can you add a test? It should fail on master and pass on this branch. Look in pandas/io for what patterns we use and place it next to similar tests

**Review by arw2019 — COMMENTED** [review]
this needs a rebase, tests, and a whatsnew

also not sure this solution is quite right atm (if the CI failures are real)

**Comment by jreback** [comment]
closing as stale. if you want to continue, pls ping and can re-open.

**Pull request closed without merging**